crypto/provider_core.c

```diff
@@ -580,6 +580,11 @@ OSSL_PROVIDER *ossl_provider_new(OSSL_LIB_CTX *libctx, const char *name,
     if (prov == NULL)
         return NULL;
 
+    if (!ossl_provider_set_module_path(prov, template.path)) {
```

Comment:
Can `template.path` ever be NULL?  If so, shouldn't you check for it to be non-NULL?

Reply:
It can.  The NULL check is already handled inside ossl_provider_set_module_path.  If the path is NULL, ossl_provider_set_module_path returns success without actually duplicating the path, which seems correct to me.

Reply:
Er, it actually does a bit more than that, it *clears* any previously set path.  Is that what you want?

https://github.com/openssl/openssl/blob/e1fd043ad7fa865a8ef9160c892b49a098d23c71/crypto/provider_core.c#L758-L767

Reply:
I believe so, yes, given that we're adding a call to ossl_provider_set_module_path with a freshly allocated provider returned from provider_new, which never sets the path value at all.

Reply:
Okie